dtfs-central-api/api-tests/v1/utilisation-reports/patch-payment.api-test.ts

```diff
@@ -0,0 +1,164 @@
+    // Arrange
+    const differentPaymentId = 25;
+    const differentPayment = PaymentEntityMockBuilder.forCurrency('GBP').withId(differentPaymentId).build();
+    await SqlDbHelper.saveNewEntry('Payment', differentPayment);
+
+    // Act
+    const response = await testApi.patch(aPatchPaymentRequestBody()).to(getUrl(reportId, differentPaymentId));
+
+    // Assert
+    expect(response.status).toBe(HttpStatusCode.NotFound);
+  });
+
+  it.each(difference(Object.values(UTILISATION_REPORT_RECONCILIATION_STATUS), [UTILISATION_REPORT_RECONCILIATION_STATUS.RECONCILIATION_IN_PROGRESS]))(
+    "returns a 400 when the report the payment is attached has the status '%s'",
+    async (status) => {
+      // Arrange
+      const report = aReport();
+      report.status = status;
+      report.feeRecords = feeRecordsForReportWithPayments(report, [payment]);
+
+      await SqlDbHelper.deleteAllEntries('UtilisationReport');
+      await SqlDbHelper.saveNewEntry('UtilisationReport', report);
+
+      // Act
+      const response = await testApi.patch(aPatchPaymentRequestBody()).to(getUrl(reportId, paymentId));
+
+      // Assert
+      expect(response.status).toBe(HttpStatusCode.BadRequest);
+    },
+  );
+
+  it('returns a 200 when the payment can be edited', async () => {
+    // Act
+    const response = await testApi.patch(aPatchPaymentRequestBody()).to(getUrl(reportId, paymentId));
+
+    // Assert
+    expect(response.status).toBe(HttpStatusCode.Ok);
+  });
+
+  it('updates the payment and report with the fields supplied in the payload', async () => {
+    // Arrange
+    const paymentAmount = 314.59;
+    const datePaymentReceived = new Date('2024-02-04');
+    const paymentReference = 'A new payment reference';
+    const tfmUserId = new ObjectId().toString();
+
+    const requestBody: PatchPaymentPayload = {
+      paymentAmount,
+      datePaymentReceived,
+      paymentReference,
+      user: {
+        ...aTfmSessionUser(),
+        _id: tfmUserId,
+      },
+    };
+
+    // Act
+    const response = await testApi.patch(requestBody).to(getUrl(reportId, paymentId));
+
+    // Assert
+    expect(response.status).toBe(HttpStatusCode.Ok);
+
+    const paymentEntity = (await SqlDbHelper.manager.findOneBy(PaymentEntity, { id: paymentId }))!;
+
+    expect(paymentEntity.amount).toBe(paymentAmount);
+    expect(paymentEntity.dateReceived).toEqual(datePaymentReceived);
+    expect(paymentEntity.reference).toBe(paymentReference);
+
+    expect(paymentEntity.lastUpdatedByTfmUserId).toBe(tfmUserId);
+    expect(paymentEntity.lastUpdatedByPortalUserId).toBeNull();
+    expect(paymentEntity.lastUpdatedByIsSystemUser).toBe(false);
+
+    const reportEntity = (await SqlDbHelper.manager.findOneBy(UtilisationReportEntity, { id: reportId }))!;
+
+    expect(reportEntity.lastUpdatedByTfmUserId).toBe(tfmUserId);
+    expect(reportEntity.lastUpdatedByPortalUserId).toBeNull();
+    expect(reportEntity.lastUpdatedByIsSystemUser).toBe(false);
```

Comment:
As we are checking db fields could we also add checks for the fee record status changing by perhaps doing a case where we go from match to does not match or the other way around? That check existing and passing would probably answer a couple of my other comments 😄